lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+ */
+'use strict';
+
+require('./browser.es6');
+require('./browser_api.es6');
+require('./web_interface.es6');
+
+foam.CLASS({
+  name: 'WebAPIs',
+  package: 'com.web.api',
+  requires: [
+    'com.web.api.Browser',
+    'com.web.api.BrowserAPI',
+    'com.web.api.WebInterface',
+    'foam.dao.EasyDAO',
+    'foam.mlang.sink.GroupBy',
+    'foam.dao.ArrayDAO',
+    'foam.mlang.ExpressionsSingleton',
+  ],
+  properties: [
+    {
+      name: 'browserAPIs',
+      documentation: `A DAO that contains pairs of browser Id
+        and interface Id.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserAPIDAO',
+          of: this.BrowserAPI,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'browsers',
+      documentation: `A DAO that contains browser names, versions
+        OS, OS versions and their browser Id.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserDAO',
+          of: this.Browser,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'interfaces',
+      documentation: `A DAO that contains interface names, API names,
+        and their interfaceId`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'InterfaceDAO',
+          of: this.WebInterface,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'mlang',
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'importAPI',
+      documentation: 'Import interface/API for a given version of browser.',
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSName',
+          documentation: `The name of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSVersion',
+          documentation: `The version of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        OSName, OSVersion, apiCatalog) {
+        let browser = this.Browser.create({
+          browserName: browserName,
+          browserVersion: browserVersion,
+          OS: OSName,
+          OSVersion: OSVersion,
+        });
+        let browserId = browser.id;
+        this.browsers.put(browser);
```

Comment:
What's the context of the call sites for this? If we find the browser is already in the DAO, should we exit early?